modules/nextflow/src/main/groovy/nextflow/executor/LatchPathFactory.groovy

```diff
@@ -40,14 +42,20 @@ class LatchPathFactory extends FileSystemPathFactory {
     @Override
     protected String getBashLib(Path target) {
         if (target.scheme != "latch") {
-            throw new ProviderMismatchException("Cannot use LatchPathFactory on non-latch path ${target.toUriString()}")
+            return null
```

Comment:
why returning null here instead of erroring?

Reply:
the top level calls every PathFactory and expects them to return null if the path passed in is not of the target scheme (ik, incredible design lol) so i have to do this garbage